[BUG] test_sorted_groupby_first_last failed in nightly tests

UCX standalone nightly tests had failures in `test_sorted_groupby_first_last`:
```
[2022-06-28T16:25:02.652Z] =========================== short test summary info ============================
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[String][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Byte][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Short][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Integer][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Long][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Float][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Double][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Boolean][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Date][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Timestamp][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Decimal(7,3)][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Decimal(12,2)][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Decimal(20,2)][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])][IGNORE_ORDER({'local': True})]
[2022-06-28T16:25:02.652Z] FAILED ../../src/main/python/hash_aggregate_test.py::test_sorted_groupby_first_last[Struct(['child0', Byte],['child1', Struct(['child0', Byte],['child1', Short],['child2', Integer],['child3', Long],['child4', Float],['child5', Double],['child6', String],['child7', Boolean],['child8', Date],['child9', Timestamp],['child10', Null])])][IGNORE_ORDER({'local': True})]
```

Sample failure:
```
[2022-06-28T16:25:02.645Z] [31m[1m____________________ test_sorted_groupby_first_last[String] ____________________[0m
[2022-06-28T16:25:02.645Z] 
[2022-06-28T16:25:02.645Z] data_gen = String
[2022-06-28T16:25:02.645Z] 
[2022-06-28T16:25:02.645Z]     @ignore_order(local=True)
[2022-06-28T16:25:02.645Z]     @pytest.mark.parametrize('data_gen', all_gen + _struct_only_nested_gens, ids=idfn)
[2022-06-28T16:25:02.645Z]     def test_sorted_groupby_first_last(data_gen):
[2022-06-28T16:25:02.645Z]         gen_fn = [('a', RepeatSeqGen(LongGen(), length=20)), ('b', data_gen)]
[2022-06-28T16:25:02.645Z]         # sort by more than the group by columns to be sure that first/last don't remove the ordering
[2022-06-28T16:25:02.645Z]         agg_fn = lambda df: df.orderBy('a', 'b').groupBy('a').agg(
[2022-06-28T16:25:02.645Z]             f.first('b'), f.last('b'), f.first('b', True), f.last('b', True))
[2022-06-28T16:25:02.645Z] >       assert_gpu_and_cpu_are_equal_collect(
[2022-06-28T16:25:02.645Z]             # First and last are not deterministic when they are run in a real distributed setup.
[2022-06-28T16:25:02.645Z]             # We set parallelism 1 to prevent nondeterministic results because of distributed setup.
[2022-06-28T16:25:02.645Z]             lambda spark: agg_fn(gen_df(spark, gen_fn, num_slices=1)),
[2022-06-28T16:25:02.645Z]             conf = {'spark.rapids.sql.explain': 'ALL'})
[2022-06-28T16:25:02.645Z] 
[2022-06-28T16:25:02.645Z] [1m[31m../../src/main/python/hash_aggregate_test.py[0m:1114: 
[2022-06-28T16:25:02.645Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-06-28T16:25:02.645Z] [1m[31m../../src/main/python/asserts.py[0m:508: in assert_gpu_and_cpu_are_equal_collect
[2022-06-28T16:25:02.645Z]     _assert_gpu_and_cpu_are_equal(func, 'COLLECT', conf=conf, is_cpu_first=is_cpu_first)
[2022-06-28T16:25:02.645Z] [1m[31m../../src/main/python/asserts.py[0m:439: in _assert_gpu_and_cpu_are_equal
[2022-06-28T16:25:02.645Z]     assert_equal(from_cpu, from_gpu)
[2022-06-28T16:25:02.645Z] [1m[31m../../src/main/python/asserts.py[0m:106: in assert_equal
[2022-06-28T16:25:02.645Z]     _assert_equal(cpu, gpu, float_check=get_float_check(), path=[])
[2022-06-28T16:25:02.645Z] [1m[31m../../src/main/python/asserts.py[0m:42: in _assert_equal
[2022-06-28T16:25:02.645Z]     _assert_equal(cpu[index], gpu[index], float_check, path + [index])
[2022-06-28T16:25:02.645Z] [1m[31m../../src/main/python/asserts.py[0m:35: in _assert_equal
[2022-06-28T16:25:02.645Z]     _assert_equal(cpu[field], gpu[field], float_check, path + [field])
[2022-06-28T16:25:02.645Z] _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
[2022-06-28T16:25:02.645Z] 
[2022-06-28T16:25:02.645Z] cpu = '\x97PÊ~$l°\x9d', gpu = 'Jór¾6JÇõ'
[2022-06-28T16:25:02.645Z] float_check = <function get_float_check.<locals>.<lambda> at 0x7fd124f39430>
[2022-06-28T16:25:02.645Z] path = [9, 'first(b)']
[2022-06-28T16:25:02.645Z] 
[2022-06-28T16:25:02.645Z]     def _assert_equal(cpu, gpu, float_check, path):
[2022-06-28T16:25:02.645Z]         t = type(cpu)
[2022-06-28T16:25:02.645Z]         if (t is Row):
[2022-06-28T16:25:02.645Z]             assert len(cpu) == len(gpu), "CPU and GPU row have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
[2022-06-28T16:25:02.645Z]             if hasattr(cpu, "__fields__") and hasattr(gpu, "__fields__"):
[2022-06-28T16:25:02.645Z]                 assert cpu.__fields__ == gpu.__fields__, "CPU and GPU row have different fields at {} CPU: {} GPU: {}".format(path, cpu.__fields__, gpu.__fields__)
[2022-06-28T16:25:02.645Z]                 for field in cpu.__fields__:
[2022-06-28T16:25:02.645Z]                     _assert_equal(cpu[field], gpu[field], float_check, path + [field])
[2022-06-28T16:25:02.645Z]             else:
[2022-06-28T16:25:02.645Z]                 for index in range(len(cpu)):
[2022-06-28T16:25:02.645Z]                     _assert_equal(cpu[index], gpu[index], float_check, path + [index])
[2022-06-28T16:25:02.645Z]         elif (t is list):
[2022-06-28T16:25:02.645Z]             assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
[2022-06-28T16:25:02.645Z]             for index in range(len(cpu)):
[2022-06-28T16:25:02.645Z]                 _assert_equal(cpu[index], gpu[index], float_check, path + [index])
[2022-06-28T16:25:02.645Z]         elif (t is tuple):
[2022-06-28T16:25:02.645Z]             assert len(cpu) == len(gpu), "CPU and GPU list have different lengths at {} CPU: {} GPU: {}".format(path, len(cpu), len(gpu))
[2022-06-28T16:25:02.645Z]             for index in range(len(cpu)):
[2022-06-28T16:25:02.645Z]                 _assert_equal(cpu[index], gpu[index], float_check, path + [index])
[2022-06-28T16:25:02.645Z]         elif (t is pytypes.GeneratorType):
[2022-06-28T16:25:02.645Z]             index = 0
[2022-06-28T16:25:02.645Z]             # generator has no zip :( so we have to do this the hard way
[2022-06-28T16:25:02.645Z]             done = False
[2022-06-28T16:25:02.645Z]             while not done:
[2022-06-28T16:25:02.645Z]                 sub_cpu = None
[2022-06-28T16:25:02.645Z]                 sub_gpu = None
[2022-06-28T16:25:02.645Z]                 try:
[2022-06-28T16:25:02.645Z]                     sub_cpu = next(cpu)
[2022-06-28T16:25:02.645Z]                 except StopIteration:
[2022-06-28T16:25:02.645Z]                     done = True
[2022-06-28T16:25:02.645Z]     
[2022-06-28T16:25:02.645Z]                 try:
[2022-06-28T16:25:02.645Z]                     sub_gpu = next(gpu)
[2022-06-28T16:25:02.645Z]                 except StopIteration:
[2022-06-28T16:25:02.645Z]                     done = True
[2022-06-28T16:25:02.645Z]     
[2022-06-28T16:25:02.645Z]                 if done:
[2022-06-28T16:25:02.645Z]                     assert sub_cpu == sub_gpu and sub_cpu == None, "CPU and GPU generators have different lengths at {}".format(path)
[2022-06-28T16:25:02.645Z]                 else:
[2022-06-28T16:25:02.645Z]                     _assert_equal(sub_cpu, sub_gpu, float_check, path + [index])
[2022-06-28T16:25:02.645Z]     
[2022-06-28T16:25:02.645Z]                 index = index + 1
[2022-06-28T16:25:02.645Z]         elif (t is dict):
[2022-06-28T16:25:02.645Z]             # The order of key/values is not guaranteed in python dicts, nor are they guaranteed by Spark
[2022-06-28T16:25:02.645Z]             # so sort the items to do our best with ignoring the order of dicts
[2022-06-28T16:25:02.645Z]             cpu_items = list(cpu.items()).sort(key=_RowCmp)
[2022-06-28T16:25:02.645Z]             gpu_items = list(gpu.items()).sort(key=_RowCmp)
[2022-06-28T16:25:02.645Z]             _assert_equal(cpu_items, gpu_items, float_check, path + ["map"])
[2022-06-28T16:25:02.645Z]         elif (t is int):
[2022-06-28T16:25:02.645Z]             assert cpu == gpu, "GPU and CPU int values are different at {}".format(path)
[2022-06-28T16:25:02.645Z]         elif (t is float):
[2022-06-28T16:25:02.645Z]             if (math.isnan(cpu)):
[2022-06-28T16:25:02.645Z]                 assert math.isnan(gpu), "GPU and CPU float values are different at {}".format(path)
[2022-06-28T16:25:02.645Z]             else:
[2022-06-28T16:25:02.645Z]                 assert float_check(cpu, gpu), "GPU and CPU float values are different {}".format(path)
[2022-06-28T16:25:02.645Z]         elif isinstance(cpu, str):
[2022-06-28T16:25:02.645Z] >           assert cpu == gpu, "GPU and CPU string values are different at {}".format(path)
[2022-06-28T16:25:02.645Z] [1m[31mE           AssertionError: GPU and CPU string values are different at [9, 'first(b)'][0m
[2022-06-28T16:25:02.645Z] 
[2022-06-28T16:25:02.645Z] [1m[31m../../src/main/python/asserts.py[0m:84: AssertionError
[2022-06-28T16:25:02.645Z] ----------------------------- Captured stdout call -----------------------------
[2022-06-28T16:25:02.645Z] ### CPU RUN ###
[2022-06-28T16:25:02.645Z] ### GPU RUN ###
[2022-06-28T16:25:02.645Z] ### COLLECT: GPU TOOK 0.43032121658325195 CPU TOOK 0.4708669185638428 ###
[2022-06-28T16:25:02.645Z] CPU OUTPUT: [Row(a=-7540734677356764604, first(b)=None, last(b)='ÿUZåU=¥\x8b', first(b)='\x00ý¦\x98vFWË', last(b)='ÿUZåU=¥\x8b'), Row(a=-5831592707909023540, first(b)=None, last(b)='þ\nIxã¢ÂJ', first(b)='\x02\x9d`º\x9c\x9976', last(b)='þ\nIxã¢ÂJ'), Row(a=-5133656973475552689, first(b)=None, last(b)='û¾¾\x92è\x14f\x8f', first(b)='\x05\x8a<879\x06k', last(b)='û¾¾\x92è\x14f\x8f'), Row(a=-4426181692283497353, first(b)=None, last(b)='þ\x1cQåê1æ{', first(b)='\x02?i/\xad\x9d\x81è', last(b)='þ\x1cQåê1æ{'), Row(a=-3917032101531217289, first(b)=None, last(b)='þïqËÉ\x15Ñ\x14', first(b)='\x01\r\x8a\x8fåÇ\x12\x0f', last(b)='þïqËÉ\x15Ñ\x14'), Row(a=-3502159106106506455, first(b)=None, last(b)='þ¢£:QÑ\x1bÎ', first(b)='\x01G\x14_J\x81MT', last(b)='þ¢£:QÑ\x1bÎ'), Row(a=-2697073954890740236, first(b)=None, last(b)='ÿË*£.m°\xad', first(b)='\x01\tÊ\x12\x04]þÕ', last(b)='ÿË*£.m°\xad'), Row(a=-2123199122092230623, first(b)=None, last(b)="ø¢'\x94\x98ó°¢", first(b)='\x04\x84Òl÷$:\r', last(b)="ø¢'\x94\x98ó°¢"), Row(a=-1, first(b)=None, last(b)='ÿ}!/\x86Ô`;', first(b)='\x03Ãt;×\x91\x90\x18', last(b)='ÿ}!/\x86Ô`;'), Row(a=207981845540287738, first(b)='\x97PÊ~$l°\x9d', last(b)='\x95³\x05\x90i\xa0·4', first(b)='\x97PÊ~$l°\x9d', last(b)='\x95³\x05\x90i\xa0·4'), Row(a=393905103838704542, first(b)=None, last(b)='ü»¹l\x18íõ\x9f', first(b)='\x04vY\x83YÍ\x00+', last(b)='ü»¹l\x18íõ\x9f'), Row(a=875130347651831881, first(b)=None, last(b)='ü\xa0U6!iß\x83', first(b)='\x03Ó_\x1fm\x08nÁ', last(b)='ü\xa0U6!iß\x83'), Row(a=4751953708995107450, first(b)=None, last(b)='ÿ_\x91\x9c|Baò', first(b)='\x00/\x12~\x1b\x94Ëq', last(b)='ÿ_\x91\x9c|Baò'), Row(a=6084712057446794809, first(b)=None, last(b)='ÿ\x83p\x06\x11>zY', first(b)="\x00Ãß\x91\x83'\x8fe", last(b)='ÿ\x83p\x06\x11>zY'), Row(a=7198729688045931692, first(b)=None, last(b)='þ\x90Û\x0bÎykC', first(b)='\x00ÐM\x0få¼V\x10', last(b)='þ\x90Û\x0bÎykC'), Row(a=7528354001793048440, first(b)=None, last(b)='ÿø<ì,\x04¸q', first(b)='\x02\x92y5\x8d_RÁ', last(b)='ÿø<ì,\x04¸q'), Row(a=7618709293599214015, first(b)=None, last(b)='ö,Ü9ìÍE$', first(b)='\x05\x8c¡\x94×Ë\x84\x0f', last(b)='ö,Ü9ìÍE$'), Row(a=7984374766242566542, first(b)=None, last(b)='þé\x9bI\x14^6-', first(b)='\x02]nÞ£ÉÎ\x9a', last(b)='þé\x9bI\x14^6-'), Row(a=9223372036854775807, first(b)=None, last(b)='ÿ\x88ì\x82\x7f\x99Òt', first(b)='\x00Î_Æ2Ñ\x16ÿ', last(b)='ÿ\x88ì\x82\x7f\x99Òt')]
[2022-06-28T16:25:02.645Z] GPU OUTPUT: [Row(a=-7540734677356764604, first(b)=None, last(b)='ÿUZåU=¥\x8b', first(b)='\x00ý¦\x98vFWË', last(b)='ÿUZåU=¥\x8b'), Row(a=-5831592707909023540, first(b)=None, last(b)='þ\nIxã¢ÂJ', first(b)='\x02\x9d`º\x9c\x9976', last(b)='þ\nIxã¢ÂJ'), Row(a=-5133656973475552689, first(b)=None, last(b)='û¾¾\x92è\x14f\x8f', first(b)='\x05\x8a<879\x06k', last(b)='û¾¾\x92è\x14f\x8f'), Row(a=-4426181692283497353, first(b)=None, last(b)='þ\x1cQåê1æ{', first(b)='\x02?i/\xad\x9d\x81è', last(b)='þ\x1cQåê1æ{'), Row(a=-3917032101531217289, first(b)=None, last(b)='þïqËÉ\x15Ñ\x14', first(b)='\x01\r\x8a\x8fåÇ\x12\x0f', last(b)='þïqËÉ\x15Ñ\x14'), Row(a=-3502159106106506455, first(b)=None, last(b)='þ¢£:QÑ\x1bÎ', first(b)='\x01G\x14_J\x81MT', last(b)='þ¢£:QÑ\x1bÎ'), Row(a=-2697073954890740236, first(b)=None, last(b)='ÿË*£.m°\xad', first(b)='\x01\tÊ\x12\x04]þÕ', last(b)='ÿË*£.m°\xad'), Row(a=-2123199122092230623, first(b)=None, last(b)="ø¢'\x94\x98ó°¢", first(b)='\x04\x84Òl÷$:\r', last(b)="ø¢'\x94\x98ó°¢"), Row(a=-1, first(b)=None, last(b)='ÿ}!/\x86Ô`;', first(b)='\x03Ãt;×\x91\x90\x18', last(b)='ÿ}!/\x86Ô`;'), Row(a=207981845540287738, first(b)='Jór¾6JÇõ', last(b)='JQÞAjÈ_C', first(b)='Jór¾6JÇõ', last(b)='JQÞAjÈ_C'), Row(a=393905103838704542, first(b)=None, last(b)='ü»¹l\x18íõ\x9f', first(b)='\x04vY\x83YÍ\x00+', last(b)='ü»¹l\x18íõ\x9f'), Row(a=875130347651831881, first(b)=None, last(b)='ü\xa0U6!iß\x83', first(b)='\x03Ó_\x1fm\x08nÁ', last(b)='ü\xa0U6!iß\x83'), Row(a=4751953708995107450, first(b)=None, last(b)='ÿ_\x91\x9c|Baò', first(b)='\x00/\x12~\x1b\x94Ëq', last(b)='ÿ_\x91\x9c|Baò'), Row(a=6084712057446794809, first(b)=None, last(b)='ÿ\x83p\x06\x11>zY', first(b)="\x00Ãß\x91\x83'\x8fe", last(b)='ÿ\x83p\x06\x11>zY'), Row(a=7198729688045931692, first(b)='bR\n¢HÔB\x12', last(b)="`cu¸»\x93x'", first(b)='bR\n¢HÔB\x12', last(b)="`cu¸»\x93x'"), Row(a=7528354001793048440, first(b)=None, last(b)='ÿø<ì,\x04¸q', first(b)='\x02\x92y5\x8d_RÁ', last(b)='ÿø<ì,\x04¸q'), Row(a=7618709293599214015, first(b)=None, last(b)='ö,Ü9ìÍE$', first(b)='\x05\x8c¡\x94×Ë\x84\x0f', last(b)='ö,Ü9ìÍE$'), Row(a=7984374766242566542, first(b)=None, last(b)='þé\x9bI\x14^6-', first(b)='\x02]nÞ£ÉÎ\x9a', last(b)='þé\x9bI\x14^6-'), Row(a=9223372036854775807, first(b)=None, last(b)='ÿ\x88ì\x82\x7f\x99Òt', first(b)='\x00Î_Æ2Ñ\x16ÿ', last(b)='ÿ\x88ì\x82\x7f\x99Òt')]
```

--- a/integration_tests/src/main/python/hash_aggregate_test.py
+++ b/integration_tests/src/main/python/hash_aggregate_test.py
@@ -1113,9 +1113,10 @@ def test_sorted_groupby_first_last(data_gen):
         f.first('b'), f.last('b'), f.first('b', True), f.last('b', True))
     assert_gpu_and_cpu_are_equal_collect(
         # First and last are not deterministic when they are run in a real distributed setup.
-        # We set parallelism 1 to prevent nondeterministic results because of distributed setup.
+        # We set parallelism and partitions to 1 to prevent nondeterministic results because
+        # of distributed setups.
         lambda spark: agg_fn(gen_df(spark, gen_fn, num_slices=1)),
-        conf = {'spark.rapids.sql.explain': 'ALL'})
+        conf = {'spark.sql.shuffle.partitions': '1'})
 
 @ignore_order
 @pytest.mark.parametrize('data_gen', all_gen, ids=idfn)

